### PR TITLE
Hosting Command Palette: Rank current site at the top

### PR DESCRIPTION
**Pull request by danielbachhuber**
Fixes https://github.com/Automattic/dotcom-forge/issues/4710

## Proposed Changes

Puts the current site at the top of the site list when using the command palette within an existing site context.

CleanShot 2023-11-30 at 14 48 49@2x

## Testing Instructions

1. Navigate to `/sites`.
2. Open the command palette, select 'Manage hosting configuration', and verify the site list matches your expected site order.
3. Navigate to Hosting Configuration for one of your sites.
4. Open the command palette, select 'Manage writing settings', and verify the current site appears at the top of the site list.

### CODE REVIEW

**Review by katinthehatsite — APPROVED** [review]
Nice work - it works as expected. If the command is available for a site and you are in the site content, the selected site will display as the first one in the nexted command.

**Review by sejas — CHANGES_REQUESTED** [review]
For me it ranks the current site to the top even if I'm on `/sites` page.

Screenshot 2023-12-01 at 10 28 48

**Comment by katinthehatsite** [comment]
> For me it ranks the current site to the top even if I'm on `/sites` page.
> 
> Screenshot 2023-12-01 at 10 28 48

Oh interesting, I am not sure why it did not rank it for me. I am wondering if perhaps the site I was testing with was already the first one in my site's list on the `/sites` page ...

**Review by sejas — COMMENTED** [review]
There is a bug that makes disappear the current site. I cannot reproduce this behaviour on `wpcalypso`

https://github.com/Automattic/wp-calypso/assets/779993/acf55668-0b04-4e6c-b696-259484bb8e28

**Steps to reproduce**
1. Navigate to a `/hosting-config/$siteSlug`
2. Open the command palette
3. Find a command like `Add new user`
4. Cancel by pressing `esc` or `backspace` to see all the root commands
5. Find a second command, it can be the same `Add new user`
6. Observe that your current site is gone from the list.

**Comment by sejas** [comment]
> Oh interesting, I am not sure why it did not rank it for me. I am wondering if perhaps the site I was testing with was already the first one in my site's list on the /sites page ...

I think Calypso live loaded directly `/sites` and you didn't have a "current/selected" site the first time you accessed.

**Review by sejas — APPROVED** [review]
I fixed the bug by removing the `splice` , which was modifying the original `sortedSites` array. https://github.com/Automattic/wp-calypso/pull/84726/commits/b6a7f4120c3996275288967de04fee49cfa81714

Working as expected. Thanks for the PR!

## Screencast


https://github.com/Automattic/wp-calypso/assets/779993/2c41b01f-0188-4fa2-b432-6b143f1af0e7

**Comment by danielbachhuber** [comment]
@sejas Thanks for the fix!

File: client/components/command-palette/use-current-site-rank-top.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import getCurrentRoute from 'calypso/state/selectors/get-current-route';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+
+export function useCurrentSiteRankTop() {
+	let currentSiteId = useSelector( getSelectedSiteId );
+	const currentPath = useSelector( ( state ) => getCurrentRoute( state ) );
+	if (
+		currentPath.startsWith( '/sites' ) ||
+		currentPath.startsWith( '/read' ) ||
+		currentPath.startsWith( '/me' )
```
Comment (danielbachhuber) [inline client/components/command-palette/use-current-site-rank-top.ts:11]: @sejas This list seems like a pain to maintain.

Instead, I think we should only return `currentSiteId` if there's a `:site` in the matched route.

What do you think?

Comment (sejas) [reply client/components/command-palette/use-current-site-rank-top.ts:11]: - Created a PR here https://github.com/Automattic/wp-calypso/pull/84829 👌

**Pull request merged**